frontend/awx/common/util/getAddedAndRemoved.ts

```diff
@@ -1,24 +1,15 @@
-import { Credential as ControllerCredential } from '../../interfaces/Credential';
-import { Label as ControllerLabel } from '../../interfaces/Label';
-import { InstanceGroup as ControllerInstanceGroup } from '../../interfaces/InstanceGroup';
-
-type Items =
-  | Pick<ControllerCredential, 'id' | 'name'>
-  | Pick<ControllerLabel, 'id' | 'name'>
-  | Pick<ControllerInstanceGroup, 'id' | 'name'>;
-
 /**
  * Utility function to get arrays containing ids of items to be removed and added
  * based on original and upsdated items
  * @param original original list of items
  * @param current updated list of items
  * @returns object containing 2 lists: ids to be added and ids to be removed
  */
-export function getAddedAndRemoved(original: Items[], current: Items[]) {
+export function getAddedAndRemoved<T extends { id: number }>(original: T[], current: T[]) {
```

Comment:
This same change is being added by @AlexSCorey here https://github.com/ansible/ansible-ui/pull/2643/files